InvenTree/InvenTree/settings.py

```diff
@@ -938,10 +938,44 @@ def _is_true(x):
 # Plugin Directories (local plugins will be loaded from these directories)
 PLUGIN_DIRS = ['plugin.builtin', ]
 
+CUSTOM_PLUGIN_DIRS = os.getenv('INVENTREE_PLUGIN_DIR', None)
+
 if not TESTING:
-    # load local deploy directory in prod
+    # load local plugin directory
     PLUGIN_DIRS.append('plugins')  # pragma: no cover
 
+    # optionally load custom plugin directory
+    if CUSTOM_PLUGIN_DIRS is not None:
+        # Allow multiple plugin directories to be specified
+        for pd in CUSTOM_PLUGIN_DIRS.split(','):
+            pd = pd.strip()
+
+            # Attempt to create the directory if it does not already exist
+            if not os.path.exists(pd):
+                try:
+                    os.makedirs(pd, exist_ok=True)
+                except Exception:
+                    logger.error(f"Could not create plugin directory '{pd}'")
+                    continue
+
+            # Ensure the directory has an __init__.py file
+            init_filename = os.path.join(pd, '__init__.py')
+
+            if not os.path.exists(init_filename):
+                try:
+                    with open(init_filename, 'w') as init_file:
+                        init_file.write("# InvenTree plugin directory\n")
+                except Exception:
+                    logger.error(f"Could not create file '{init_filename}'")
+                    continue
+
+            if os.path.exists(pd) and os.path.isdir(pd):
```

Comment:
Suggestion: `pathlib.Path` is a bit easier to read and handles linux / mac / win paths better- should we switch to it for this? I would be glad to provide a patch for code.

Reply:
@matmair yes please add a patch to this branch